test/rules/void_checks.dart

```diff
@@ -112,7 +112,7 @@ async_function() {
 
 inference() {
   f(void Function() f) {}
-  f(() //LINT
+  f(() // OK
```

Comment:
I'd prefer to keep a LINT here. I think it's fine to relax for method/function references but not for lambda.